hw/xbox/mcpx_apu.c

```diff
@@ -411,9 +411,9 @@ static void scratch_rw(hwaddr sge_base, unsigned int max_sge,
         unsigned int entry = (addr + i) / TARGET_PAGE_SIZE;
         assert(entry < max_sge);
```

Comment:
There might be an off-by-one here. My 2017 code changed this to `assert(entry <= max_sge);`. I might add another commit for this, but I'm not sure if I can find the time to debug / test this - scratch access is a bit wonky (because it depends on DMA transfers).